Lantstool - FT

eclipseeer opened the pull request:
- Added examples for primitives - FT tokens
- Updated linkdrop components

eclipseeer commented:
@race-of-sloths

race-of-sloths commented:
@eclipseeer Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!
Are you going to win race-of-sloths this month? If so, you should speed up! Run, Sloth, run!

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://race-of-sloths.com/images/bot-banner-dark.svg">
    <source media="(prefers-color-scheme: light)" srcset="https://race-of-sloths.com/images/bot-banner-light.svg">
    <img alt="Shows inviting banner with latest news." src="https://race-of-sloths.com/images/bot-banner-light.svg">
</picture>](https://race-of-sloths.com/latest-news)

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/eclipseeer?type=bot&pr=near/docs/2473&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/eclipseeer?type=bot&pr=near/docs/2473&theme=light">
    <img alt="Shows profile picture for the author of the PR" src="https://badge.race-of-sloths.com/eclipseeer?type=bot&pr=near/docs/2473&theme=white">
</picture>](https://race-of-sloths.com/profile/eclipseeer)


<details><summary>Current status: <i>waiting for scoring</i></summary>

We're waiting for maintainer to score this pull request with `@race-of-sloths score [0,1,2,3,5,8,13]` command. Alternatively, autoscoring [1,2] will be applied for this pull request

</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/eclipseeer)

For maintainers:
- Score pull requests that participate in the Race of Sloths and receive a reward
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Invite contributor** `@race-of-sloths invite` to invite the contributor to participate in a race or include it, if it's already a runner.
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

bucanero commented on docs/2.build/5.primitives/ft.md:
````diff
@@ -70,9 +78,15 @@ On initialization you will define the token's metadata such as its name (e.g. Et
-```
+<Tabs groupId="code-tabs">
+  <TabItem value="🖥️ Cargo-NEAR" label="🖥️ Cargo-NEAR">
+    <CargoNearCreateTokenManually />
````
In this case I'd leave the `cargo` code here, like:

```suggestion

```bash
cargo near deploy build-non-reproducible-wasm <account-id> with-init-call new json-args '{"owner_id": "<owner-account>", "total_supply": "1000000000000000", "metadata": { "spec": "ft-1.0.0", "name": "Example Token Name", "symbol": "EXLT", "decimals": 8 }}' prepaid-gas '100.0 Tgas' attached-deposit '0 NEAR' network-config testnet sign-with-keychain send
```

```

eclipseeer replied:
I would not do this because:
1. It breaks the general pattern where we use components inside the tabs. Mixing two approaches makes the documentation look inconsistent, making it harder for other contributors to understand when to use components and when to keep them in the file.
2. It complicates editing and reading because it eliminates spaces, making it difficult to recognize the nesting.

eclipseeer replied:
@bucanero also I would recommend the format this command and improve the readability:

```bash
cargo near deploy build-non-reproducible-wasm <account-id> \
  with-init-call new \
  json-args '{
    "owner_id": "<owner-account>",
    "total_supply": "1000000000000000",
    "metadata": {
      "spec": "ft-1.0.0",
      "name": "Example Token Name",
      "symbol": "EXLT",
      "decimals": 8
    }
  }' \
  prepaid-gas '100.0 Tgas' \
  attached-deposit '0 NEAR' \
  network-config testnet \
  sign-with-keychain send
```

P.S We can use AI for such tasks.

bucanero replied:
yes, it looks better for this `cargo near` command 👍

bucanero commented on website/src/components/docs/primitives/ft/cargo-near/create-manually.md:
In this case I'd leave this snippet in the original article's markdown file.

bucanero commented on docs/2.build/5.primitives/ft.md:
````diff
@@ -70,9 +78,15 @@ On initialization you will define the token's metadata such as its name (e.g. Et
-cargo near deploy build-non-reproducible-wasm <account-id> with-init-call new json-args '{"owner_id": "<owner-account>", "total_supply": "1000000000000000", "metadata": { "spec": "ft-1.0.0", "name": "Example Token Name", "symbol": "EXLT", "decimals": 8 }}' prepaid-gas '100.0 Tgas' attached-deposit '0 NEAR' network-config testnet sign-with-keychain send
-```
+<Tabs groupId="code-tabs">
+  <TabItem value="🖥️ Cargo-NEAR" label="🖥️ Cargo-NEAR">
````
```suggestion
  <TabItem value="🖥️ CLI" label="🖥️ CLI">
```

bucanero reviewed (changes requested):
hi @eclipseeer 👋  Thanks for the PR 🙏 

Please check my comments and I'll merge the PR 👍

bucanero commented on website/src/components/docs/primitives/ft/cargo-near/create-manually.md:
````diff
@@ -0,0 +1,3 @@
+```bash
+cargo near deploy build-non-reproducible-wasm <account-id> with-init-call new json-args '{"owner_id": "<owner-account>", "total_supply": "1000000000000000", "metadata": { "spec": "ft-1.0.0", "name": "Example Token Name", "symbol": "EXLT", "decimals": 8 }}' prepaid-gas '100.0 Tgas' attached-deposit '0 NEAR' network-config testnet sign-with-keychain send
````
```suggestion
cargo near deploy build-non-reproducible-wasm <account-id> \
  with-init-call new \
  json-args '{
    "owner_id": "<owner-account>",
    "total_supply": "1000000000000000",
    "metadata": {
      "spec": "ft-1.0.0",
      "name": "Example Token Name",
      "symbol": "EXLT",
      "decimals": 8
    }
  }' \
  prepaid-gas '100.0 Tgas' \
  attached-deposit '0 NEAR' \
  network-config testnet \
  sign-with-keychain send
```

eclipseeer commented:
As we agreed to keep all CLi command (near-cli-rs and cargo-near) under the same tab, I moved the file to near-cli folder and removed the cargo-near folder
